File: packages/go/events/user_events.go
package events

import "time"

// UserEventType represents user lifecycle events
type UserEventType string

const (
	UserCreated UserEventType = "user.created"
	UserUpdated UserEventType = "user.updated"
	UserDeleted UserEventType = "user.deleted"
)

// UserCreatedPayload is emitted when a new user registers
type UserCreatedPayload struct {
	UserID    string    `json:"userId"`
	Email     string    `json:"email"`
	FirstName string    `json:"firstName"`
	LastName  string    `json:"lastName"`
	Role      string    `json:"role"`
	CreatedAt time.Time `json:"createdAt"`
}

// UserUpdatedPayload is emitted when a user profile is updated
type UserUpdatedPayload struct {
	UserID    string     `json:"userId"`
	Email     string     `json:"email"`
	FirstName string     `json:"firstName"`
	LastName  string     `json:"lastName"`
	Role      string     `json:"role"`
	IsActive  bool       `json:"isActive"`
	CreatedAt time.Time  `json:"createdAt"`
	UpdatedAt time.Time  `json:"updatedAt"`
	AvatarID  *string    `json:"avatarMediaId,omitempty"`
}

// UserDeletedPayload is emitted when a user is deleted
type UserDeletedPayload struct {
	UserID    string    `json:"userId"`
	DeletedAt time.Time `json:"deletedAt"`
}

File: packages/ts/src/events/user.events.ts
/**
 * User lifecycle events shared by services.
 */
export enum UserEventType {
  Created = 'user.created',
  Updated = 'user.updated',
  Deleted = 'user.deleted',
}

export interface UserCreatedPayload {
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  role: string;
  createdAt: string;
}

export interface UserUpdatedPayload {
  userId: string;
  email: string;
  firstName: string;
  lastName: string;
  role: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  avatarMediaId?: string;
}

export interface UserDeletedPayload {
  userId: string;
  deletedAt: string;
}

export interface UserEventEnvelope<T = unknown> {
  eventId: string;
  eventType: UserEventType | string;
  traceId?: string;
  timestamp: number;
  payload: T;
}
